src/OrchardCore/OrchardCore.Workflows.Abstractions/Models/Workflow.cs

```diff
@@ -31,12 +31,27 @@ public class Workflow
         public WorkflowStatus Status { get; set; }
         public string FaultMessage { get; set; }
 
+        /// <summary>
+        /// The timeout in milliseconds to acquire a lock before executing this workflow instance.
+        /// </summary>
+        public int LockTimeout { get; set; }
```

Comment:
My initial thoughts was

is this only if running distributed?
In which case we should only see it if we are running distributed.
But then I see that it is also about atomicity when not distributed from your initial comments above.

Am I correct there @jtkech ?

In which case we just need to document (another pr, I know you don't write docs ;) what it all means for both an ordinary single instance site, and / or a distributed site.

Reply:
@deanmarcussen 

> is this only if running distributed?

No, also if the current distributed lock implementation is a local lock and / or if there is only one running instance, this because concurrent requests / WF events may still happen, **but only if both locking times are > 0**.

Note: In my PR related to background tasks atomicity, it's not the same because for a given running instance a background task can't be executed concurrently

> Am I correct there @jtkech ?

Yes, it's not exclusive to a distributed env, it just uses the current distributed lock implementation that may be a distributed / local lock.

> In which case we just need to document (another pr, I know you don't write docs ;) what it all means for both an ordinary single instance site, and / or a distributed site.

;)